rotkehlchen/chain/ethereum/modules/balancer/decoder.py

```diff
@@ -0,0 +1,296 @@
+            amount=amount_out,
+            asset=to_token,
+        )
+        action_item = ActionItem(
+            action='skip & keep',
+            sequence_index=tx_log.log_index,
+            from_event_type=HistoryEventType.RECEIVE,
+            from_event_subtype=HistoryEventSubType.NONE,
+            asset=to_token,
+            amount=to_amount,
+            to_event_type=None,
+            to_event_subtype=None,
+            to_counterparty=CPT_BALANCER_V2,
+            to_notes=None,
+            extra_data={
+                'from_token': from_token_address,
+                'amount_in': amount_in,
+            },
+        )
+
+        # When ETH is swapped it is wrapped to WETH and the ETH transfer happens before the SWAP
+        # event, we need to detect it we haven't done it yet.
+        if len(action_items) == 0 and from_token_address == self.weth.evm_address:
+            # when swapping eth the transfer event appears before the V2_SWAP event so we need
+            # to check if the asset swapped was ETH or not.
+            amount_of_eth = asset_normalized_value(
+                amount=amount_in,
+                asset=self.eth,
+            )
+            for event in decoded_events:
+                if (
+                    event.asset == A_ETH and event.balance.amount == amount_of_eth and
+                    event.event_type == HistoryEventType.SPEND and
+                    event.event_subtype == HistoryEventSubType.NONE
+                ):
+                    event.event_type = HistoryEventType.TRADE
+                    event.event_subtype = HistoryEventSubType.SPEND
+                    event.notes = f'Swap {event.balance.amount} {self.eth.symbol} in Balancer v2 from {event.location_label}'  # noqa: E501
+                    event.counterparty = CPT_BALANCER_V2
+
+        return None, [action_item]
+
+    def _maybe_enrich_balancer_v2_transfers(  # pylint: disable=no-self-use
+            self,
+            token: 'EvmToken',  # pylint: disable=unused-argument
+            tx_log: EvmTxReceiptLog,
+            transaction: EvmTransaction,
+            event: HistoryBaseEntry,
+            action_items: list[ActionItem],
+            all_logs: list[EvmTxReceiptLog],  # pylint: disable=unused-argument
+    ) -> bool:
+        """
+        Enrich tranfer transactions to account for swaps in balancer v2 protocol.
+        May raise:
+        - UnknownAsset
+        - WrongAssetType
+        """
+        if action_items is None or len(action_items) == 0 or transaction.to_address != VAULT_ADDRESS:  # noqa: E501
+            return False
+
+        if action_items[-1].extra_data is None:
+            return False
+
+        asset = event.asset.resolve_to_evm_token()
+
+        if (
+            isinstance(action_items[-1].asset, EvmToken) is False or
+            action_items[-1].asset.evm_address != tx_log.address or  # type: ignore[attr-defined]  # noqa: E501 It fails to detect the previous comprobation
```

Comment:
```suggestion
            action_items[-1].asset.evm_address != tx_log.address or  # type: ignore[attr-defined]  # noqa: E501 It fails to detect the previous MUMBO JUMBO
```

> comprobation

I googled this. It's a real word but I bet you not even the late queen's grandma knows what it means. If you explain something in a comment use plain English. I google it and I still don't get what you want to say. When you write a comment you are ELI5ing to a kid.

Also if it's too long you can always write it above the condition where you type ignored.

You can check with me in discord about the explanation.